### PR TITLE
BUG: [OmniSciDB] Fix TopK when used as filter

### LINKED ISSUE #1918: BUG: TopK used as filter is not working on OmniSciDB backend
TopK is not working on OmniSciDB/MapD backend. eg.:

```
>>> alltypes[alltypes.string_col.topk(3)].count().compile()

---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-39-95fa1f84c573> in <module>
----> 1 alltypes[alltypes.string_col.topk(3)].count().compile()

/mnt/sda1/dev/quansight/ibis-project/ibis/ibis/expr/types.py in compile(self, limit, params)
    213         from ibis.client import compile
    214 
--> 215         return compile(self, limit=limit, params=params)
    216 
    217     def verify(self):

/mnt/sda1/dev/quansight/ibis-project/ibis/ibis/client.py in compile(expr, limit, params, **kwargs)
    288 def compile(expr, limit=None, params=None, **kwargs):
    289     backend, = validate_backends(list(find_backends(expr)))
--> 290     return backend.compile(expr, limit=limit, params=params, **kwargs)
    291 
    292 

/mnt/sda1/dev/quansight/ibis-project/ibis/ibis/client.py in compile(self, expr, params, limit)
    199         """
    200         query_ast = self._build_ast_ensure_limit(expr, limit, params=params)
--> 201         return query_ast.compile()
    202 
    203     def _build_ast_ensure_limit(self, expr, limit, params=None):

/mnt/sda1/dev/quansight/ibis-project/ibis/ibis/sql/compiler.py in compile(self)
     52     def compile(self):
     53         compiled_setup_queries = [q.compile() for q in self.setup_queries]
---> 54         compiled_queries = [q.compile() for q in self.queries]
     55         compiled_teardown_queries = [
     56             q.compile() for q in self.teardown_queries

/mnt/sda1/dev/quansight/ibis-project/ibis/ibis/sql/compiler.py in <listcomp>(.0)
     52     def compile(self):
     53         compiled_setup_queries = [q.compile() for q in self.setup_queries]
---> 54         compiled_queries = [q.compile() for q in self.queries]
     55         compiled_teardown_queries = [
     56             q.compile() for q in self.teardown_queries

/mnt/sda1/dev/quansight/ibis-project/ibis/ibis/sql/compiler.py in compile(self)
   1565 
   1566         # FROM, JOIN, UNION
-> 1567         from_frag = self.format_table_set()
   1568 
   1569         # WHERE

/mnt/sda1/dev/quansight/ibis-project/ibis/ibis/sql/compiler.py in format_table_set(self)
   1682 
   1683         helper = self.table_set_formatter(self, self.table_set)
-> 1684         fragment += helper.get_result()
   1685 
   1686         return fragment

/mnt/sda1/dev/quansight/ibis-project/ibis/ibis/mapd/compiler.py in get_result(self)
    143 
    144         if isinstance(op, ops.Join):
--> 145             self._walk_join_tree(op)
    146         else:
    147             self.join_tables.append(self._format_table(self.expr))

/mnt/sda1/dev/quansight/ibis-project/ibis/ibis/sql/compiler.py in _walk_join_tree(self, op)
   1793         self._validate_join_predicates(op.predicates)
   1794 
-> 1795         jname = self._get_join_type(op)
   1796 
   1797         # Read off tables and join predicates left-to-right in

/mnt/sda1/dev/quansight/ibis-project/ibis/ibis/sql/compiler.py in _get_join_type(self, op)
   1834 
   1835     def _get_join_type(self, op):
-> 1836         return self._join_names[type(op)]
   1837 
   1838     def _quote_identifier(self, name):

KeyError: <class 'ibis.expr.operations.LeftSemiJoin'>

```

### LINKED ISSUE #2129: [Parquet] ParquetClient.table "path" parameter default value
Checking the method ParquetClient.table

```
    def table(self, name, path):
        if name not in self.list_tables(path):
            raise AttributeError(name)

        if path is None:
            path = self.root

        # get the schema
        f = path / "{}.parquet".format(name)

        parquet_file = pq.ParquetFile(str(f))
        schema = sch.infer(parquet_file.schema)

        table = self.table_class(name, schema, self).to_expr()
        self.dictionary[name] = f

        return table
```

it seems `path` should have a default value `None`. Similar approach used by `CSVClient.table`: https://github.com/ibis-project/ibis/blob/master/ibis/file/csv.py#L57

### PR DESCRIPTION
**Pull request by xmnlab**
Resolve #1918 . Additionally fix #2129

### CODE REVIEW

File: ibis/omniscidb/compiler.py [hunk 1]
```diff
@@ -180,6 +180,7 @@ class OmniSciDBTableSetFormatter(compiles.TableSetFormatter):
     _join_names = {
         ops.InnerJoin: 'JOIN',
         ops.LeftJoin: 'LEFT JOIN',
+        ops.LeftSemiJoin: 'JOIN',  # needed by topk as filter
```
Comment (xmnlab) [inline ibis/omniscidb/compiler.py]: not sure if it is the best approach ... but at least it is working now.
maybe #512 would be a better approach to fix this issue

File: ibis/file/parquet.py [hunk 2]
```diff
@@ -75,7 +78,9 @@ def insert(self, path, expr, **kwargs):
         table = pa.Table.from_pandas(df)
         pq.write_table(table, str(path))
 
-    def table(self, name, path):
+    def table(
+        self, name: str, path: Optional[str] = None
+    ) -> ibis.expr.types.TableExpr:
```
Comment (jreback) [inline ibis/file/parquet.py]: use the standard ibis import

import ibis.expr.types as ir (at the top)

File: ibis/file/parquet.py [hunk 3]
```diff
@@ -75,7 +78,9 @@ def insert(self, path, expr, **kwargs):
         table = pa.Table.from_pandas(df)
         pq.write_table(table, str(path))
 
-    def table(self, name, path):
+    def table(
+        self, name: str, path: Optional[str] = None
```
Comment (jreback) [inline ibis/file/parquet.py]: why is this needed?

File: ibis/omniscidb/compiler.py (hunk 1 above)
Comment (jreback) [reply ibis/omniscidb/compiler.py]: that's a really old issue, has things changed in the interim?

Comment (xmnlab) [reply ibis/omniscidb/compiler.py]: it seems it was not changed. but maybe we need to check this better.

**Comment by xmnlab** [comment]
release note added and ci green.

**Comment by jreback** [comment]
lgtm. tip on release notes; don't put them at the top, rather insert between others so to avoid conflicts.

**Comment by jreback** [comment]
thanks @xmnlab

**Comment by xmnlab** [comment]
thanks @jreback for the review!
